interactions/api/models/message.py

```diff
@@ -154,1186 +155,1088 @@ class ChannelMention(DictSerializerMixin):
     """
-    A class object representing the footer of an embed.
 
-    The structure of an embed footer:
+    id: Snowflake = field(converter=Snowflake)
+    channel_id: Snowflake = field(converter=Snowflake)
+    guild_id: Optional[Snowflake] = field(converter=Snowflake, default=None)
+    author: User = field(converter=User, add_client=True, default=None)
+    member: Optional[Member] = field(
+        converter=converters.optional(Member), default=None, add_client=True
+    )
+    content: str = field(default=None)
+    timestamp: datetime = field(converter=datetime.fromisoformat, default=None)
+    edited_timestamp: Optional[datetime] = field(converter=datetime.fromisoformat, default=None)
+    tts: bool = field(default=None)
+    mention_everyone: bool = field(default=None)
+    # mentions: array of Users, and maybe partial members
+    mentions: Optional[List[Union[Member, User]]] = field(
+        default=None
+    )  # todo convert to the right types
+    mention_roles: Optional[List[str]] = field(default=None)
+    mention_channels: Optional[List[ChannelMention]] = field(
+        converter=convert_list(ChannelMention), default=None
+    )
+    attachments: List[Attachment] = field(converter=convert_list(Attachment), default=None)
+    embeds: List[Embed] = field(converter=convert_list(Embed), default=None)
+    reactions: Optional[List[ReactionObject]] = field(
+        converter=convert_list(ReactionObject), default=None
+    )
+    nonce: Optional[Union[int, str]] = field(default=None)
+    pinned: bool = field(default=None)
+    webhook_id: Optional[Snowflake] = field(converter=Snowflake, default=None)
+    type: int = field(default=None)
+    activity: Optional[MessageActivity] = field(converter=MessageActivity, default=None)
+    application: Optional[Application] = field(converter=Application, default=None)
+    application_id: Optional[Snowflake] = field(converter=Snowflake, default=None)
+    message_reference: Optional[MessageReference] = field(converter=MessageReference, default=None)
+    flags: int = field(default=None)
+    referenced_message: Optional[MessageReference] = field(converter=MessageReference, default=None)
+    interaction: Optional[MessageInteraction] = field(
+        converter=MessageInteraction, default=None, add_client=True
+    )
+    thread: Optional[Channel] = field(converter=Channel, default=None, add_client=True)
 
-    .. code-block:: python
+    components: Optional[Union["Component", List["Component"]]] = field(default=None)  # noqa: F821
+    sticker_items: Optional[List[PartialSticker]] = field(converter=convert_list, default=None)
+    stickers: Optional[List[Sticker]] = field(
+        converter=convert_list(Sticker), default=None
+    )  # deprecated
 
-        interactions.EmbedFooter(
-            text="yo mama so short, she can fit in here",
-        )
+    def __repr__(self) -> str:
+        return self.content
```

Comment:
I *think* this repr can removed - `attrs` can just use it's usual repr. Using the content isn't *really* great for representing messages when many people won't even have the guild message content intent on.

I know, this kind of sounds like a thing for a separate PR, and it honestly might be, but this was likely here in the first place because there was no auto-generating repr. There is one now.